### PR TITLE
Wait on daemon down

### PR DESCRIPTION
**Pull request by pascal-fischer**
## Describe your changes
This PR awaits the engine shutdown before sending the confirmation back to the cli. It also fixes an issue where there was a race condition when running multiple up and down commands in sequence.

### Checklist
- [x] Is it a bug fix
- [ ] Is a typo/documentation fix
- [ ] Is a feature enhancement
- [ ] It is a refactor
- [ ] Created tests that fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: client/internal/networkmonitor/monitor_bsd.go [hunk 1]
```diff
@@ -21,19 +22,28 @@ func checkChange(ctx context.Context, nexthopv4, nexthopv6 systemops.Nexthop, ca
...
+
 	for {
 		select {
 		case <-ctx.Done():
 			return ErrStopped
 		default:
 			buf := make([]byte, 2048)
 			n, err := unix.Read(fd, buf)
-			if err != nil {
+			if err != nil && !errors.Is(err, unix.EBADF) && !errors.Is(err, unix.EINVAL) {
```
Comment (lixmal) [inline client/internal/networkmonitor/monitor_bsd.go]: I think we need to `continue` on the ignored errors as well, just don't log them, or leg them on debug. Apologies for the previous suggestion

Comment (pascal-fischer) [reply client/internal/networkmonitor/monitor_bsd.go]: fixed

File: client/server/server.go [hunk 2]
```diff
@@ -593,7 +593,27 @@ func (s *Server) Down(_ context.Context, _ *proto.DownRequest) (*proto.DownRespo
...
+	for {
+		select {
+		case <-ctx.Done():
+			return &proto.DownResponse{}, nil
+		default:
+			if !engine.IsWGIfaceUp() {
+				return &proto.DownResponse{}, nil
+			}
+
+			if time.Since(startTime) > maxWaitTime {
```
Comment (lixmal) [inline client/server/server.go]: I think it's more idomatic to use `time.After` here and include the returned channel in the `select`, unless you still need to check `if !engine.IsWGIfaceUp()` in any case?

File: client/internal/engine.go [hunk 3]
```diff
@@ -1533,3 +1533,19 @@ func isChecksEqual(checks []*mgmProto.Checks, oChecks []*mgmProto.Checks) bool {
...
 	})
 }
+
+func (e *Engine) IsWGIfaceUp() bool {
+	if e == nil || e.wgInterface == nil {
+		return false
+	}
+	iface, err := net.InterfaceByName(e.wgInterface.Name())
+	if err != nil {
+		return false
```
Comment (lixmal) [inline client/internal/engine.go:1559]: Please log the error to ease debugging, at least on `debug` level

**Review by lixmal — APPROVED** [review]

**Pull request merged**